riscv/src/compiler.rs

```diff
@@ -1416,44 +1297,29 @@ fn process_instruction(instr: &str, args: &[Argument], coprocessors: &CoProcesso
             vec![format!("x1 <== jump_dyn({rs});")]
         }
         "call" | "tail" => {
-            // Depending on what symbol is called, the call is replaced by a
-            // powdr-asm call, or a call to a coprocessor if a special function
-            // has been recognized.
             assert_eq!(args.len(), 1);
             let label = &args[0];
-            let replacement = match label {
-                Argument::Expression(Expression::Symbol(l)) => {
-                    try_coprocessor_substitution(l, coprocessors)
-                }
-                _ => None,
-            };
-            match (replacement, instr) {
-                (None, instr) => {
-                    let arg = argument_to_escaped_symbol(label);
-                    let dest = if instr == "tail" { "tmp1" } else { "x1" };
-                    vec![format!("{dest} <== jump({arg});")]
-                }
-                // Both "call" and "tail" are pseudo-instructions that are
-                // supposed to use x6 to calculate the high bits of the
-                // destination address. Our implementation does not touch x6,
-                // but no sane program would rely on this behavior, so we are
-                // probably fine.
-                (Some(replacement), "call") => vec![replacement],
-                (Some(replacement), "tail") => {
-                    vec![replacement, "tmp1 <== jump_dyn(x1);".to_string()]
-                }
-                (Some(_), _) => unreachable!(),
-            }
```

Comment:
Why is it fine to remove this?

Reply:
Before, special calls were done by replacing `call/tail special_func` with special handing code.
Now these are `ecall` for specific syscalls